ext/gd/php_gd.h

```diff
@@ -77,6 +77,31 @@ PHP_MINIT_FUNCTION(gd);
 PHP_MSHUTDOWN_FUNCTION(gd);
 PHP_RSHUTDOWN_FUNCTION(gd);
 
+#include <gd.h>
```

Comment:
> In file included from main/internal_functions_cli.c:45:
> /home/vsts/work/1/s/ext/gd/php_gd.h:80:10: fatal error: gd.h: No such file or directory

I think having an include at this position means we need to pollute the global include path with the gd incdir, rather than just for the gd extension.

Reply:
I don't think we actually need to export `php_gd_image_object` and `php_gd_exgdimage_from_zobj_p` though. Just `php_gd_libgdimageptr_from_zval_p` as a non-inline function should be enough. If you change the return value to `struct gdImageStruct *` we don't need the gd.h include either.

Reply:
I'm afraid all this is necessary, so we can get the offsetof of the std member in a portable way. :(

Reply:
Why is the offsetof needed?

Reply:
To be clear, what I means is: If the implementation is part of the .c file and only exported via PHP_GD_API, why do we need to expose the offset?

Reply:
Ah, understood now. Thanks! Will update the PR this evening.